Source/Request.swift

```diff
@@ -1528,11 +1528,11 @@ public class DownloadRequest: Request {
     @Protected
     private var mutableDownloadState = DownloadRequestMutableState()
 
-    /// If the download is resumable and eventually cancelled, this value may be used to resume the download using the
-    /// `download(resumingWith data:)` API.
+    /// If the download is resumable and is eventually cancelled or fails, this value may be used to resume the download
+    /// using the `download(resumingWith data:)` API.
     ///
     /// - Note: For more information about `resumeData`, see [Apple's documentation](https://developer.apple.com/documentation/foundation/urlsessiondownloadtask/1411634-cancel).
-    public var resumeData: Data? { mutableDownloadState.resumeData }
+    public var resumeData: Data? { mutableDownloadState.resumeData ?? error?.downloadResumeData }
```

Comment:
Return the `error?.downloadResumeData` in computed property may not be a good solution. It will still get wrong data when other developers use `mutableDownloadState` internally. In my view, a better solution is to assign `error?.downloadResumeData`  directly to `mutableDownloadState`.

Reply:
Nobody should be doing that, as it's internal to the library.

Reply:
Sorry, I didn't make it clear. The other developers mean the contributor of Alamofire, if they wanna use `mutableDownloadState` in the `Request.swifit`, they may get the wrong return.

Reply:
That value is documented to be the value from cancellation, and the documentation of the property should make it clear it can come from two sources, so I don't think there's an issue.